src/connectors/configure/connectConfigure.js

```diff
@@ -91,13 +91,18 @@ export default function connectConfigure(renderFn = noop, unmountFn = noop) {
       },
 
       removeSearchParameters(state) {
-        // widgetParams are assumed 'controlled',
-        // so they override whatever other widgets give the state
-        return state.mutateMe(mutableState => {
-          Object.keys(widgetParams.searchParameters).forEach(key => {
-            delete mutableState[key];
-          });
-        });
+        // We leverage the Helper internals to remove the `widgetParams` from
+        // the state. The function `setQueryParameters` omits the values that
+        // are `undefined` on the next state.
```

Comment:
I'm not sure I understand what you mean.
Is it that for example

```js
const state = { a: 'something', b: 'someotherthing' }
const nextState = state.setQueryParameters({ a: undefined })
// nextState is { b: 'someotherthing' }
```

Reply:
I tried writing a test for it and it fails.
https://github.com/algolia/algoliasearch-helper-js/compare/drafts/setQueryParameters

Reply:
you need to compare to the `next` branch, it already passes & has a test there: 

https://github.com/algolia/algoliasearch-helper-js/blob/ffa13f45dd4a71d0a148ecd72fa1da6fb8da4c4f/test/spec/SearchParameters/setQueryParameters.js#L77-L91

https://github.com/algolia/algoliasearch-helper-js/compare/next...drafts/setQueryParameters